### ISSUE TITLE
Foundry 0.8.9 fails to render in the electron window on Ubuntu 22.04

### ISSUE THREAD

**Issue by aaclayton** [foundryvtt/foundryvtt#7135]
Originally in GitLab by @moorepants

I upgraded from Ubuntu 20.04 to Ubuntu 22.04 and I started up Foundry 0.8.9 (I've been using Foundry for 1.5 years on 20.04) and the electron window opens but no ui is displayed (only a blank black content window). I can still access Foundry via a web browser and everything works fine there.

In the terminal I get a couple of suspicious messages:

```
$ /opt/foundryvtt/foundryvtt --trace-warnings
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
FoundryVTT | 2022-05-15 16:37:15 | [info] Running on Node.js - Version 14.16.0
FoundryVTT | 2022-05-15 16:37:15 | [info] Foundry Virtual Tabletop - Version 0.8.9
```

and

```
(node:8902) UnhandledPromiseRejectionWarning: Error: ERR_FAILED (-2) loading 'http://localhost:30000/'
    at rejectAndCleanup (electron/js2c/browser_init.js:217:1457)
    at Object.stopLoadingListener (electron/js2c/browser_init.js:217:1832)
    at Object.emit (events.js:315:20)
    at Object.EventEmitter.emit (domain.js:467:12)
(Use `foundryvtt --trace-warnings ...` to show where the warning was created)
(node:8902) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org /api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:8902) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I reinstalled Foundry 0.8.9 and have the same issue.

These are the relevant electron and chromium versions that foundry displays:

```
FoundryVTT | 2022-05-15 17:06:11 | [info] Running Electron application with version 12.2.2 using Chromium 89.0.4389.128
```

There is more useful information in this reddit post I made: https://www.reddit.com/r/FoundryVTT/comments/uq85w1/foundry_no_longer_working_in_ubuntu_2204/

Some key things are that:

1) Chromium switched from a deb package to a snap.
2) The libva error is reported in other forums and is associated with chromium being a snap.
3) The libva error goes away if I set this environment variable `export LIBVA_DRIVER_NAME=amdgpu`, but the `UnhandledPromiseRejectionWarning: Error: ERR_FAILED (-2)` remains.

**Comment by anathemamask** [comment]
Given the fact this only appears to impact 0.8.9 and not newer versions, any fix for this would involve updating the version of electron packaged with 0.8.9 which.....at that point, the solution is to recommend upgrading to V9+. This is a #wontfix. Sorry.

**Comment by moorepants** [comment]
I was finally able to upgrade to Foundry 9 now that all modules I use work with v9. I'm still seeing this same issue (blank electron window content). Here is what the terminal reports:

```
mesa: for the --simplifycfg-sink-common option: may only occur zero or one times!
mesa: for the --global-isel-abort option: may only occur zero or one times!
mesa: for the --amdgpu-atomic-optimizations option: may only occur zero or one times!
FoundryVTT | 2022-07-16 19:27:51 | [info] Running on Node.js - Version 14.16.0
FoundryVTT | 2022-07-16 19:27:51 | [info] Foundry Virtual Tabletop - Version 9 Build 269
FoundryVTT | 2022-07-16 19:27:51 | [info] Loading data from user directory - 
FoundryVTT | 2022-07-16 19:27:51 | [info] Application Options:
{
  "port": 30000,
  "upnp": true,
  "fullscreen": false,
  "hostname": null,
  "localHostname": null,
  "routePrefix": null,
  "sslCert": null,
  "sslKey": null,
  "awsConfig": null,
  "dataPath": "",
  "passwordSalt": null,
  "proxySSL": false,
  "proxyPort": null,
  "serviceConfig": null,
  "updateChannel": "stable",
  "language": "en.core",
  "upnpLeaseDuration": null,
  "compressStatic": true,
  "world": null,
  "isElectron": true,
  "isNode": false,
  "isSSL": false,
  "noupdate": false,
  "noIPDiscovery": false,
  "demo": null,
  "adminKey": "****************"
}
FoundryVTT | 2022-07-16 19:27:51 | [info] Software license verification succeeded
FoundryVTT | 2022-07-16 19:27:51 | [info] Requesting UPnP port forwarding to destination port 30000
FoundryVTT | 2022-07-16 19:27:51 | [info] Running Electron application with version 12.2.2 using Chromium 89.0.4389.128
FoundryVTT | 2022-07-16 19:27:51 | [info] Server started and listening on port 30000
FoundryVTT | 2022-07-16 19:27:52 | [info] Created client session 
FoundryVTT | 2022-07-16 19:27:52 | [warn] Package pf2e-dragruler contains 2 compendium definitions that still use the deprecated 'entity' field instead of 'type'.
FoundryVTT | 2022-07-16 19:27:52 | [warn] Package spellcaster-utility-pf1 contains 1 compendium definitions that still use the deprecated 'entity' field instead of 'type'.
FoundryVTT | 2022-07-16 19:27:52 | [warn] Package spellcaster-utility-pf1 is using the old "systems" manifest key where a "system" key is expected
(node:27359) UnhandledPromiseRejectionWarning: Error: ERR_FAILED (-2) loading 'http://localhost:30000/'
    at rejectAndCleanup (electron/js2c/browser_init.js:217:1457)
    at Object.stopLoadingListener (electron/js2c/browser_init.js:217:1832)
    at Object.emit (events.js:315:20)
    at Object.EventEmitter.emit (domain.js:467:12)
(Use `foundryvtt --trace-warnings ...` to show where the warning was created)
(node:27359) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:27359) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by moorepants** [comment]
It is interesting to note that Foundry v9 reports the same Electron and Chromium versions as my prior messages for v0.8.9.

**Comment by plutoneld** [comment]
I think you need to provide some more information about how you are launching FVTT, as well as double checking your install locations.

The logs indicates you are missing a dataPath, and I'm not aware of a situation where that should be the case.

I just tested downloading v9 on Ubuntu 22.04, extracting the zip into a new folder, and launching it successfully first by double clicking the foundryvtt executable and then through terminal.

**Comment by moorepants** [comment]
> The logs indicates you are missing a dataPath, and I'm not aware of a situation where that should be the case.

I deleted that from the log for privacy reasons.

> I just tested downloading v9 on Ubuntu 22.04, extracting the zip into a new folder, and launching it successfully first by double clicking the foundryvtt executable and then through terminal.

I updated Foundry via the inbuilt updater. So that is one difference.

**Comment by plutoneld** [comment]
Ah, the built-in updater doesn't update the Electron version, amongst other things. Try a fresh install of v9.

Iirc i think the node version was up to 16 on my fresh install.

**Comment by moorepants** [comment]
I downloaded a fresh Foundry 9.269 and the electron app works correctly again. The versions of the underlying software are:

- Node.js 16.5.0
- Electron 15.3.5
- Chromium 94.0.4606.81

I believe I tried this same thing with Foundry 0.8.9 but a fresh download did not correct this issue, so that is likely still outstanding for users of 0.8.*.

This seems to have confused a number of users, so maybe either 1) disable updating the software in the application or 2) ensure the updater updates *all* of the software or 3), at the minimum, warn that updating in the application doesn't actually update the software fully.

Thanks for the help.
